### DISCUSSION TITLE
Delete playlist. How

### DISCUSSION THREAD

**Discussion by CH-1967** [hklages/node-red-contrib-sonos-plus#272] [Q&A]
Hello I am not getting anywhere with this.

I set a playlist on a single Sonos speaker with group.play.sonosplaylist.
This is played.
Now I want to remove/delete this playlist from the speaker again.
I have already tried a few things. But it just does not work.
Can someone tell me how I can implement this.

**Comment by hklages** [comment]
Hi. Playlists are stored in the household not on a single player. Have a look at [household remove playlist](https://github.com/hklages/node-red-contrib-sonos-plus/wiki/A.1-Universal-Node#householdremovesonosplaylist)  
Does that help?

**Comment by CH-1967** [comment]
Sorry. I must have expressed myself a little unclearly.
I do not want to delete the playlist itself.
I just want to remove it from the playlist.
![26-10-_2023_08-24-10](https://github.com/hklages/node-red-contrib-sonos-plus/assets/50620532/679a799b-2579-4166-a21c-6bf67a957d7a)

Here is a picture of it.

**Comment by hklages** [comment]
Ah. Damit wir das selbe Verständnis haben: Es gibt die "Queue", das ist das was die SONOS App anzeigt. 
Diese kann man als "Playlist" speichern und dann über den Namen auch wieder in die "Queue" laden. 
Abgespielt wird immer das, was in der "Queue" ist - entweder sequentiell oder zufällig, je nachdem wie man die "Queue" eingestellt hat. 

In der "Queue" kann man einzelne Songs mit dem Befehl "[group.remove.tracks](https://github.com/hklages/node-red-contrib-sonos-plus/wiki/A.1-Universal-Node#groupremovetracks)" entfernen. Allerdings ändert das nicht die ursprüngliche "Playlist".  

Will man die "Playlist" ändern könnte man das erreichen, indem man erst in der "Queue" den Song entfernt, dann die "Playlist" löscht und unter dem selben Namen die "Queue" als "Playlist" speichert. Achtung beim Löschen einen Songs:  Wenn die "Queue" im Modus "zufällig" ist, ist die Songnummer eine andere. 

Stand heute 2023-10-26 habe ich keinen Befehl, mit dem man die einzelnen Songs einer "Playlist" entfernen kann. Falls die obige Lösung nicht passt, schaue ich mir Doku an und prüfe, ob SONOS eine Funktionalität bereit stellt um das zu implementieren.

**Comment by CH-1967** [comment]
Das mit dem Befehl group.remove.tracks ist das was ich suchte. 
Ich möchte einfach die ganze Queue löschen.

An den Playliste möchte ich keine Änderung vornehmen.

**Comment by hklages** [comment]
Wenn du die ganze "Queue" löschen willst, geht auch "group.clear.queue".

**Comment by CH-1967** [comment]
Super. group.remove.tracks  ist das was ich wollte.
Ich sehe aber das ich mich da noch besser einlesen muss damit ich verstehe was / wie gemacht werden kann.
Danke.

Beschäftige mich erst seit ca. 3 Tagen mit Node-Red und es ist erstaunlich was da alles Umgesetzt werden kann.

**Comment by CH-1967** [comment]
Ich schliesse meine Anfrage da mir sehr schnell geholfen wurde. Danke vielmals.
I close my request because I was helped very quickly. Thank you very much.
